GoogleInstagram/app/src/main/java/com/example/cynchen/googleinstagram/AllImagesFragment.java

```diff
@@ -0,0 +1,181 @@
+        saved_images_all = new ArrayList<String>(dbhelper.readData());
+        k = 0; //initialize k to be 0
+
+        //First step, check if there is anything in saved_images_all, or else the app will crash trying to
+        //index from something with a size 0
+        if (saved_images_all.size() == 0){
+                webView_all.loadUrl("about:blank"); //nothing: put blank webview
+        }
+        else {
+            webView_all.loadUrl(saved_images_all.get(0)); //something: show the first image
+        }
+
+        //If next button is clicked, there are different scenarios that could happen
+        //One: 0 images saved -> alert dialog notifying user to save an image pops up
+        //Two: 1 image saved -> do nothing, keep loading that one image
+        //Three: k is at the last image of the list -> go to the 0th image
+        //Four: All other k index values -> load the (k+1)th image
+        nextbutton_all = (Button) rootView.findViewById(R.id.next_image_all);
+        nextbutton_all.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (saved_images_all.size() == 0) {
+                    create_alert_all_images(); //alert dialog creation function
+                }
+                else if (saved_images_all.size() == 1){
+                    webView_all.loadUrl(saved_images_all.get(k));
+                }
+                else if (saved_images_all.size() - 1 == k){
+                    k = 0;
+                    webView_all.loadUrl(saved_images_all.get(k));
+                }
+                else {
+                    k += 1;
+                    webView_all.loadUrl(saved_images_all.get(k));
+                }
+            }
+        });
+
+        //If previous button is clicked, there are different scenarios that could happen:
+        //One: 0 images saved -> popup alert dialog again
+        //Two: 1 image saved -> do nothing, keep loading that one image
+        //Three: k is at the first image of the list -> go to the last image
+        //Four: All other k index values -> load the (k-1)th image
+        previousbutton_all = (Button) rootView.findViewById(R.id.previous_image_all);
+        previousbutton_all.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (saved_images_all.size() == 0){
+                    create_alert_all_images(); //alert dialog creation function
+                }
+                else if (saved_images_all.size() == 1){
+                    webView_all.loadUrl(saved_images_all.get(k));
+                }
+                else if (k == 0){
+                    k = saved_images_all.size()-1;
+                    webView_all.loadUrl(saved_images_all.get(k));
+                }
+                else {
+                    k -= 1;
+                    webView_all.loadUrl(saved_images_all.get(k));
```

Comment:
This and the next button could be consolidated a lot. Anytime you feel like you are writing the same code as you already have for more than a couple lines, look into consolidating.